scripts/performance_benchmark.py

```
@@ -0,0 +1,394 @@
+        print("Testing memory usage patterns...")
+
+        try:
+            import psutil
+            import os
+
+            process = psutil.Process(os.getpid())
+            initial_memory = process.memory_info().rss / 1024 / 1024  # MB
+
+            # Make many requests to test memory leaks
+            test_data = {
+                "name": f"Memory Test {i}",
```

Ruff (0.13.1)

Undefined name `i`

(F821)